examples/lang/erc721/src/lib.rs

```diff
@@ -0,0 +1,275 @@
+            self.mint(env, &to, &id)?;
+            env.println(&format!("Erc721::minted(token = {:?}) = {:?}", id, to));
+            Ok(())
+        }
+    }
+
+    impl Erc721 {
+        fn approve(&mut self, env: &EnvHandler, to: &AccountId, id: &TokenId) -> Result<(), &'static str> {
+            let caller = env.caller();
+            if self.owner_of(id) != caller {
+                return Err("not owner")
+            };
+
+            self.token_approvals
+                .insert(*id, caller)
+                .ok_or("cannot approve token")?;
+            env.emit(Approval {
+                owner: caller,
+                to: *to,
+                id: *id,
+            });
+            Ok(())
+        }
+
+        fn transfer_from(&mut self, env: &EnvHandler, from: &AccountId, to: &AccountId, id: &TokenId) -> Result<(), &'static str> {
+            let caller = env.caller();
+            if !self.approved_or_owner(&caller, id){
+                return Err("not approved")
+            };
+
+            self.clear_approval(id)?;
+            self.remove_token_from(from, id)?;
+            self.add_token_to(to, id)?;
+            env.emit(Transfer {
+                from: *from,
+                to: *to,
+                id: *id,
+            });
+            Ok(())
+        }
+
+        fn clear_approval(&mut self, id: &TokenId) -> Result<(), &'static str> {
+            if !self.token_approvals.contains_key(id){
+                return Err("approval not found");
+            };
+
+            match self.token_approvals.remove(id) {
+                Some(_res) => Ok(()),
+                None => Err("cannot remove token approval")
+            }
+        }
+
+        fn remove_token_from(&mut self, from: &AccountId, id: &TokenId) -> Result<(), &'static str> {
+            if self.owner_of(id) != *from {
+                return Err("not owner")
+            };
+            if !self.exists(id){
+                return Err("token not found");
+            };
+
+            self.decrease_counter_of(from)?;
+            self.token_owner
+                .remove(id)
+                .ok_or("cannot remove token")?;
+            Ok(())
+        }
+
+        fn add_token_to(&mut self, to: &AccountId, id: &TokenId) -> Result<(), &'static str> {
+            if self.exists(id){
+                 return Err("token exists and assigned")
+            };
+
+            self.increase_counter_of(to);
+            self.token_owner
+                .insert(*id, *to)
+                .ok_or("cannot insert token")?;
+            Ok(())
+        }
+
+        fn mint(&mut self, env: &EnvHandler, to: &AccountId, id: &TokenId) -> Result<(), &'static str> {
+            if *to == AccountId::from([0x0; 32]){
+                return Err("cannot mint to account zero")
+            };
+            if self.exists(id){
+                return Err("token already minted")
+            };
+
+            self.token_owner
+                .insert(*id, *to)
```

Comment:
@Robbepop any idea why it fails to insert a token? It can do the next steps and insert a counter but the token fails.

Reply:
I can't figure out the reason but if I comment the next line `.ok_or("cannot insert token")?;` it just works